airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/documentation/documentation.py

```diff
@@ -0,0 +1,605 @@
+            connector=connector,
+            message=f"User facing documentation file {connector.documentation_file_path} exists",
+        )
+
+
+class CheckDocumentationContent(DocumentationCheck):
+    """
+    For now, we check documentation structure for sources with sl >= 300.
+    """
+
+    applies_to_connector_languages = [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]
+    applies_to_connector_ab_internal_sl = 300
+    applies_to_connector_types = ["source"]
+
+
+class CheckDocumentationLinks(CheckDocumentationContent):
+    name = "Links used in connector documentation are valid"
+    description = f"The user facing connector documentation should update invalid links in connector documentation."
+    valid_status_codes = [200, 403, 401, 405, 429, 503]  # we skip 4xx due to needed access
+
+    def validate_links(self, connector: Connector) -> List[str]:
+        errors = []
+        threads = []
+
+        def request_link(docs_link: str) -> None:
+            try:
+                response = requests.get(docs_link)
+                if response.status_code not in self.valid_status_codes:
+                    errors.append(f"{docs_link} with {response.status_code} status code")
+            except requests.exceptions.SSLError:
+                pass
+            except requests.exceptions.ConnectionError:
+                pass
+
+        for link in DocumentationContent(connector=connector).links:
+            process = Thread(target=request_link, args=[link])
+            process.start()
+            threads.append(process)
+
+        for process in threads:
+            process.join(timeout=30)  # 30s timeout for process else link will be skipped
+            process.is_alive()
+
+        return errors
+
+    def _run(self, connector: Connector) -> CheckResult:
+        if not connector.documentation_file_path or not connector.documentation_file_path.exists():
+            return self.fail(
+                connector=connector,
+                message="Could not check documentation links as the documentation file is missing.",
+            )
+
+        if not connector.documentation_file_path.read_text().rstrip():
+            return self.fail(
+                connector=connector,
+                message="Documentation file is empty",
+            )
+
+        errors = self.validate_links(connector)
+        if errors:
+            return self.fail(
+                connector=connector,
+                message=f"Connector documentation uses invalid links: {', '.join(errors)}",
+            )
+        return self.pass_(
+            connector=connector,
+            message="Documentation links are valid",
+        )
+
+
+class CheckDocumentationHeadersOrder(CheckDocumentationContent):
+    name = "Connectors documentation headers structure, naming and order follow our guidelines"
+
+    CONNECTOR_SPECIFIC_HEADINGS = "<Connector-specific features>"
+
+    @property
+    def description(self) -> str:
```

Comment:
I'm not sure to get how dynamic this description is. Can we store it inside a static file?

Reply:
it only adds section headers and content to description of the check from template. so if template was changed it will be automatically added in description.